### PR TITLE
MAX31856 enhancements

### PR DESCRIPTION
**Pull request by ba58smith**
- Change the conversion mode to ONESHOT.
- Add a second constructor, per @mairas request.
- Remove potential endless loop with begin().
- Add a constructor without the obsolete data ready pin.
- Update the example.

### CODE REVIEW

File: src/sensors/max31856_thermocouple.cpp [hunk 1]
```diff
@@ -11,22 +11,35 @@ MAX31856Thermocouple::MAX31856Thermocouple(int8_t cs_pin, int8_t mosi_pin,
     : NumericSensor(config_path),
       data_ready_pin_{drdy_pin},
       read_delay_{read_delay} {
-  load_configuration();
+  load_configuration();      
   max31856_ = new Adafruit_MAX31856(cs_pin, mosi_pin, miso_pin, clk_pin);
   if (!max31856_->begin()) {
-    while (1) delay(10);
+    debugW("No MAX31856 detected: check wiring.");
```
Comment (mairas) [inline src/sensors/max31856_thermocouple.cpp:17]: It seems to me that if the physical sensor is not detected, the measurement loop in `enable()` is started anyway. I wonder what `readThermocoupleTemperature()` returns if the device is not present? My preference would be to only start the repeat loop if the `begin()` call was successful.

**Review by mairas — COMMENTED** [review]
Very nice. One small comment/request.

File: src/sensors/max31856_thermocouple.cpp (hunk 1 above)
Comment (ba58smith) [reply src/sensors/max31856_thermocouple.cpp:18]: Done. Look OK?

**Comment by ba58smith** [comment]
@BjarneBitscrambler, I thought about your suggestion to allow the sensor to "recover", but I couldn't think of any scenario in which it would actually do what it's intended to. The actual MAX31856 is almost certainly going to be soldered in place, or at least solidly plugged into a socket, and should be protected from accidental physical disturbances that could unseat it. If it comes unseated, it will have to be physically reseated, and at that point, it would make sense to just push the RESET button on the ESP, which starts the code all over again. IOW, I couldn't think of a situation in which `begin()` would fail that can be fixed easily enough to try to make the code handle it.

**Comment by BjarneBitscrambler** [comment]
@ba58smith - thinking it through as you describe, it does sound unlikely a mechanical/physical failure would occur that could be corrected by just re-initializing. So your present approach seems appropriate.

At the risk of letting more worms loose, I noticed in the MAX31856 spec sheet that it can detect various faults, like broken thermocouple wires. Any thoughts on detecting that and outputting a `null` instead of a number?  I'll totally agree if you say it goes quite a bit beyond basic sensor functionality...

**Comment by ba58smith** [comment]
@BjarneBitscrambler, I looked at the Datasheet, and see the open circuit and over/under voltage detection. They both look doable, but I'm going to say either or both of those could be done as an enhancement, in a new PR, later.

BTW, after a discussion w/ @mairas, it was determined that, at least for now, invalid sensor output should simply stop, rather than sending `null`, with an appropriate debugE message indicating why output is not happening. That's what I've done in this PR.

File: src/sensors/max31856_thermocouple.cpp (hunk 1 above)
Comment (mairas) [reply src/sensors/max31856_thermocouple.cpp:18]: Yup!

**Comment by mairas** [comment]
Looks good to me now!

**Pull request merged**